### DISCUSSION TITLE
Question: Coding guidelines compliance

### DISCUSSION THREAD

**Discussion by felixhao28** [dotnet/roslyn#56243] [Q&A] [answered]
I randomly came across this piece of code:

https://github.com/dotnet/roslyn/blob/480ea44fd12b2be5b25518353aa731dc9d96863a/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/SourceTextExtensions_SharedWithCodeStyle.cs#L36:L42

Is it normal/recommended to use preprocessor directives of conditional compilation to include code in comments?

**Comment by CyrusNajmabadi** [comment, marked as answer]
I wrote that code. It's normal :-)

**Reply by felixhao28** [reply]
Thanks for your replyng. May I ask what's the merit of using this, instead of simply commenting out that part of the code?

**Reply by CyrusNajmabadi** [reply]
I'm representing a block of code.  I don't want comments there are it can be ambiguous if the comments are part of the example or not.  Put another way, i'd prefer this code to be left-aligned at column 0.

**Reply by felixhao28** [reply]
Is it a personal preference or an agreed C# coding style used in Roslyn or Microsoft in general?

**Reply by CyrusNajmabadi** [reply]
It got checked in. So it's accepted :-)

**Reply by jasonmalinowski** [reply]
For the record, I only see @CyrusNajmabadi doing this though. :smile:

**Reply by jmarolf** [reply]
I would argue @CyrusNajmabadi's case is a little special. He is discussing how code it parsed and needs to show, as plainly as possible, what the code looks like to the parser. I do not see this being the case for _any_ code outside of this repo and I can't really recommend it as a general approach for showing code in comments though its use in roslyn is appropriate.
